solr/core/src/java/org/apache/solr/schema/TextField.java

```diff
@@ -165,6 +167,20 @@ public Query getFieldTermQuery(QParser parser, SchemaField field, String externa
     return new TermQuery(new Term(field.getName(), br));
   }
 
+  @Override
+  public Query getPrefixQuery(QParser parser, SchemaField sf, String termStr) {
+    final var query = super.getPrefixQuery(parser, sf, termStr);
```

Comment:
Why override these methods in TextField & StrField with duplicative code when you could modify `FieldType.getPrefixQuery` instead?  Doing it there also allows the cool existence query thing.

Reply:
I see that the limit doesn't make much sense for those two yet it's also harmless (I think).  I prefer a consistent approach with no duplication.  Thanks for doing this.

Reply:
Well there is _some_ harm, albeit small.  A prefix query on a "known safe" field type like Enum might trigger the limit, and cause the "Admin aggravation/annoyance" we've discussed and tried to minimize throughout this PR.

That'd require a user to submit a query like `my_enum_field:i*`, which is presumably a pretty rare case.  But you can imagine a user maybe doing that as a shorthand way to get docs matching a few target enum values